### DISCUSSION TITLE
Access hyper body non-destructively

### DISCUSSION THREAD

**Discussion by dspicher** [hyperium/hyper#2907] [Q&A] [answered]
I have a middleware function that takes a `mut req: axum::http::request::Request<hyper::Body>`. In there, I need to compute a function over the body serialization (i.e. I need essentially a `body: &[u8]`).

In order for the downstream handlers to still have access to the body, I'm doing something like this currently:
```rust
let body_bytes = hyper::body::to_bytes(req.body_mut()).await.unwrap();
// ... compute function over `body_bytes`
*req.body_mut() = body_bytes.into(); // re-insert body for downstream handlers
let mut resp = next.run(req).await;
```

Is there a smarter way of doing this, that doesn't require the manual body re-insertion?

**Comment by seanmonstar** [comment, marked as answer]
No, what you've done works. The body from the network is streamed, it can't be replayed naturally. So the only way is to "record" it, and give that to the next layer.

**Reply by dspicher** [reply]
Alright, thanks for the reply! :+1:
